.travis.yml

```diff
@@ -17,7 +17,7 @@ before_install:
 install:
 - conda create -n test-environment python=$TRAVIS_PYTHON_VERSION
 - source activate test-environment
-- conda install numpy matplotlib scipy pandas nose pip ipython=5.1
+- conda install numpy matplotlib scipy pandas networkx nose pip ipython=5.1
```

Comment:
i think you need this new requirement in many places.  alternatively, you could do an import in circuit with a try/except error message to install networkx

Reply:
OK. Only the method checking if the network is all connected (no open port) and the graph plot require `networkx`. If the latter is not present, it will raise an `ImportError` error informing this is required.